### PR TITLE
feat(avodiance_by_lc): make it possible to configurate flexibly

### PR DESCRIPTION
**Pull request by satoshi-ota**
## Description

Previously, normal avoidance module & avoidance by lane change module use common parameters and activation logic.

In this PR, I make it possible to configurate avoidance target for each modules.

```yaml
      target_object:
        car:
          is_target: true                              # [-]
          execute_num: 2                               # [-]
          moving_speed_threshold: 1.0                  # [m/s]
          moving_time_threshold: 1.0                   # [s]
          max_expand_ratio: 0.0                        # [-]
          envelope_buffer_margin: 0.3                  # [m]
          safety_buffer_lateral: 0.7                   # [m]
```

Please approve :arrow_down: before this PR.
https://github.com/autowarefoundation/autoware_launch/pull/469

## Tests performed

- [x] [PASS TIER IV INTERNAL SCENARIOS
](https://evaluation.tier4.jp/evaluation/reports/829ef700-1de9-56d3-8d9e-648c6db08340?project_id=prd_jt)
## Effects on system behavior

Improve target filtering logic.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: planning/behavior_path_planner/src/scene_module/lane_change/avoidance_by_lane_change.cpp [hunk 1]
```diff
@@ -34,13 +34,26 @@ AvoidanceByLaneChange::AvoidanceByLaneChange(
...
+      const auto target_class = utils::getHighestProbLabel(object.object.classification) == p.first;
+      return target_class && object.avoid_required;
+    });
+
+    if (num >= p.second.execute_num) {
+      is_more_than_threshold = true;
+    }
+  }
+
+  if (!is_more_than_threshold) {
```
Comment (zulfaqar-azmi-t4) [inline planning/behavior_path_planner/src/scene_module/lane_change/avoidance_by_lane_change.cpp:56]:
```suggestion
  const auto is_less_than_threshold =
    std::any_of(object_parameters.begin(), object_parameters.end(), [&](const auto & p) {
      const auto & objects = avoidance_data_.target_objects;
      const size_t num = std::count_if(objects.begin(), objects.end(), [&p](const auto & object) {
        const auto target_class =
          utils::getHighestProbLabel(object.object.classification) == p.first;
        return target_class && object.avoid_required;
      });
      return num < p.second.execute;
    });

  if (!is_less_than_threshold) {
```

@satoshi-ota 
This part over here seems suitable to use `std::any_of`, so this is my suggestion (I am sorry for changing some part). 
Looking forward to hear you thought in this.

File: planning/behavior_path_planner/src/scene_module/lane_change/avoidance_by_lane_change.cpp [hunk 2]
```diff
@@ -179,7 +186,7 @@ ObjectData AvoidanceByLaneChange::createObjectData(
   const auto object_closest_index = findNearestIndex(path_points, object_pose.position);
   const auto object_closest_pose = path_points.at(object_closest_index).point.pose;
   const auto t = utils::getHighestProbLabel(object.classification);
-  const auto object_parameter = p->object_parameters.at(t);
+  const auto object_parameter = avoidance_parameters_->object_parameters.at(t);
```
Comment (zulfaqar-azmi-t4) [inline planning/behavior_path_planner/src/scene_module/lane_change/avoidance_by_lane_change.cpp:189]:
```suggestion
  const auto & object_parameter = avoidance_parameters_->object_parameters.at(t);
```

Suggest adding reference.

File: planning/behavior_path_planner/src/scene_module/lane_change/avoidance_by_lane_change.cpp (hunk 1 above)
Comment (satoshi-ota) [reply planning/behavior_path_planner/src/scene_module/lane_change/avoidance_by_lane_change.cpp]: Fixed in https://github.com/autowarefoundation/autoware.universe/pull/4258/commits/2e2c965a02c4a61cd80818e1565be6f0a707c88c.

File: planning/behavior_path_planner/src/scene_module/lane_change/avoidance_by_lane_change.cpp (hunk 2 above)
Comment (satoshi-ota) [reply planning/behavior_path_planner/src/scene_module/lane_change/avoidance_by_lane_change.cpp]: Fixed in https://github.com/autowarefoundation/autoware.universe/pull/4258/commits/2e2c965a02c4a61cd80818e1565be6f0a707c88c.

**Review by zulfaqar-azmi-t4 — APPROVED** [review]
LGTM

**Pull request merged**